purchase_propagate_qty/models/purchase.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2014-2016 Numérigraphe SARL
+# Copyright 2017 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, api
+from odoo.tools import float_is_zero
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    @api.multi
+    def write(self, values):
+        res = super().write(values)
+        for line in self:
+            if line.state != 'purchase':
+                continue
+            for move in line.move_dest_ids | line.move_ids:
+                if move.state in ('cancel', 'done'):
+                    continue
+                if move.product_id != line.product_id:
```

Comment:
do you need also check for uom? there can be 2 same products with different uom'

Reply:
@i-vyshnevska I don't think so: I'm checking the stock moves related to the purchase line -> 1. they should have the same uom and 2. I'm working with the fields which are converted in the product's default unit, so I believe I'm safe.